### ISSUE TITLE
SEARCH - display only nodes matching the search query (hide others)

### ISSUE THREAD

**Issue by szilderus** [frontend-collective/react-sortable-tree#245]
I'm using SEARCH -> nodes are highlighted - Is there a possibility to hide all other leaves? 
When I have very large tree and I'm searching if particular node - all others that do not match the search query are obscuring the view. (imagine that I have tens or hundreds nodes that do not fit into one screen - it is not easy to identify parents)

Thank you

**Comment by lifejuggler** [comment]
@szilderus  you can kind of get around this issue by utilizing `searchFinishCallback` and storing the filtered values in a separate state from the treeData and replace treeData with the filteredDate if there is filtered values

**Comment by wuweiweiwu** [comment]
@szilderus @fritz-c Ill work on this over the weekend :)

**Comment by wuweiweiwu** [comment]
@szilderus @lifejuggler added in #256!

**Comment by szilderus** [comment]
Thanks for adding this but what I've ment is something little bit different - imagine for example that you have 'needle' in the list of 1000 'hay' elements that are ON THE SAME LEVEL - you need to scroll 1000 elements to view the 'needle'. Is it possible to HIDE all 'hay' elements? And leave on the tree only 'needles'. Display only branches that contains 'needle'. 

Regards!

**Comment by owcc** [comment]
@szilderus yeah, **hide-no-matching-node** not **close-no-matching-node**, this is really useful for tree search with large nodes, I have more than 8000 nodes in my project

**Comment by subeax** [comment]
Hi Guys! Do you have any plans/updates on this feature? It's pretty useful.

**Comment by piotrkochan** [comment]
Any update?
